src/libmongoc/src/mongoc/mongoc-scram-private.h

```diff
@@ -123,6 +123,348 @@ _mongoc_sasl_prep_required (const char *str);
 char *
 _mongoc_sasl_prep (const char *in_utf8, int in_utf8_len, bson_error_t *err);
 
+/* returns how many bytes a UTF-8 character is. */
+size_t
+_mongoc_utf8_char_length (const uint8_t *c);
+
+/* returns how many characters are in a UTF-8 string. Returns -1 on error. */
+size_t
+_mongoc_utf8_string_length (const uint8_t *s);
+
+/* returns whether a UTF-8 character is valid or not. */
+bool
+_mongoc_utf8_is_valid (const uint8_t *c, size_t length);
+
+/* returns whether a character is between two limits (inclusive). */
+bool
+_mongoc_code_unit_between_code_unit (const uint8_t c,
+                                     const uint8_t lower,
+                                     const uint8_t upper);
+
+/* returns whether a codepoint exists in the specified table. The table format
+ * is that the 2*n element is the lower bound and the 2*n + 1 is the upper bound
+ * (both inclusive). */
+bool
+_mongoc_is_code_point_in_table (uint32_t code,
```

Comment:
```suggestion
_mongoc_utf8_code_point_is_in_table (uint32_t code,
```
Suggest renaming for consistency with other UTF-8 functions.